viewer/packages/nodes-api/src/NodesCdfClient.ts

```diff
@@ -76,27 +76,36 @@ export class NodesCdfClient implements NodesApiClient {
     return { treeIndex: ancestor.treeIndex, subtreeSize: ancestor.subtreeSize };
   }
 
-  async getBoundingBoxByNodeId(
+  async getBoundingBoxesByNodeIds(
     modelId: CogniteInternalId,
     revisionId: CogniteInternalId,
-    nodeId: CogniteInternalId,
-    box?: THREE.Box3
-  ): Promise<THREE.Box3> {
-    const response = await this._client.revisions3D.retrieve3DNodes(modelId, revisionId, [{ id: nodeId }]);
-    if (response.length < 1) {
-      throw new Error('NodeId not found');
-    }
-    const boundingBox3D = response[0].boundingBox;
-    if (boundingBox3D === undefined) {
-      throw new Error(`Node ${nodeId} doesn't have a defined bounding box`);
-    }
+    nodeIds: CogniteInternalId[]
+  ): Promise<THREE.Box3[]> {
+    const chunks = chunkInputItems(nodeIds, NodesCdfClient.MaxItemsPerRequest);
+    const mappedBoundingBoxPromises = [...chunks].map(async chunk => {
+      return this._client.revisions3D.retrieve3DNodes(
+        modelId,
+        revisionId,
+        chunk.map(id => {
+          return { id };
+        })
+      );
+    });
 
-    const min = boundingBox3D.min;
-    const max = boundingBox3D.max;
-    const result = box || new THREE.Box3();
-    result.min.set(min[0], min[1], min[2]);
-    result.max.set(max[0], max[1], max[2]);
-    return result;
+    const mappedBoundingBoxes = await Promise.all(mappedBoundingBoxPromises);
+    const resultBoxes = mappedBoundingBoxes
+      .flat()
+      .filter(node => node.boundingBox)
+      .map(node => {
+        const result = new THREE.Box3();
+        const min = node.boundingBox!.min;
+        const max = node.boundingBox!.max;
+        result.min.set(min[0], min[1], min[2]);
+        result.max.set(max[0], max[1], max[2]);
```

Comment:
Seems like this could have been replaced with the new `toThreeBox3` function you added in this PR

Reply:
viewer/packages/utilities/src/three/toThreeBox3.ts